src/diffusers/video_processor.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Union
+
+import numpy as np
+import PIL
+import torch
+
+from .image_processor import VaeImageProcessor
+
+
+class VideoProcessor:
+    """Simple video processor."""
+
+    @staticmethod
+    def tensor2vid(video: torch.Tensor, processor: "VaeImageProcessor", output_type: str = "np"):
+        """Converts a video tensor to a list of frames for export."""
+        batch_size, channels, num_frames, height, width = video.shape
+        outputs = []
+        for batch_idx in range(batch_size):
+            batch_vid = video[batch_idx].permute(1, 0, 2, 3)
+            batch_output = processor.postprocess(batch_vid, output_type)
+            outputs.append(batch_output)
+
+        if output_type == "np":
+            outputs = np.stack(outputs)
+        elif output_type == "pt":
+            outputs = torch.stack(outputs)
+        elif not output_type == "pil":
+            raise ValueError(f"{output_type} does not exist. Please choose one of ['np', 'pt', 'pil']")
+
+        return outputs
+
+    @staticmethod
+    def preprocess_video(video: List[Union[PIL.Image.Image, np.ndarray, torch.Tensor]]):
```

Comment:
video is just a list of frames, so `VaeImageProcessr.preprocess` should be able to process one video as it is
it is possible the video input is a list of videos, in that case, we can use `preprocess` to preprocess each video in the list; so this function should 3 things
1. determine if it is a list of videos, if not, make it a list of one video
2. loop through the list and use preprocess on each 
3. list of tensor -> tensor